liquidation-monitor/src/agent.js

```diff
@@ -0,0 +1,500 @@
+
+    if (data.borrow[tokenAddress] === undefined) {
+      data.borrow[tokenAddress] = {};
+    }
+    if (data.supply[tokenAddress] === undefined) {
+      data.supply[tokenAddress] = {};
+    }
+    /* eslint-enable no-param-reassign */
+
+    ts(
+      `Now tracking ${data.tokens[tokenAddress].symbol
+      } with ${data.tokens[tokenAddress].tokenDecimals
+      } decimals at address ${tokenAddress}`,
+    );
+  }
+}
+// #endregion
+
+// Initializes data required for handler
+function provideInitialize(data) {
+  return async function initialize() {
+    // #region Assign configurable fields
+    /* eslint-disable no-param-reassign */
+    data.alertMinimumIntervalSeconds = config.alertMinimumIntervalSeconds;
+    data.protocolName = config.protocolName;
+    data.protocolAbbreviation = config.protocolAbbreviation;
+    data.developerAbbreviation = config.developerAbbreviation;
+    data.alert = config.liquidationMonitor.alert;
+    data.minimumLiquidationInUSD = config.liquidationMonitor.triggerLevels.minimumLiquidationInUSD;
+    data.cTokenABI = getAbi('cErc20.json');
+    data.provider = getEthersProvider();
+    data.accounts = {}; // Health of all accounts, calcHealth, lastUpdated, [assetsIn addresses]?
+    data.supply = {}; // qty of cTokens (not Tokens)
+    data.borrow = {}; // qty of Tokens (not cTokens)
+    data.tokens = {}; // each cToken's address, symbol, contract, ratio, price, lastUpdatePrice
+    data.newAccounts = []; // New account from transaction events
+    data.totalNewAccounts = 0;
+
+    // Compound API filter and Comptroller contract
+    const { maximumHealth, minimumBorrowInETH } = config.liquidationMonitor.triggerLevels;
+    const {
+      comptrollerAddress, maxTrackedAccounts,
+      oracleAddress, oneInchAddress,
+    } = config.liquidationMonitor;
+    const comptrollerABI = getAbi(config.liquidationMonitor.comptrollerABI);
+    const oracleABI = getAbi(config.liquidationMonitor.oracleABI);
+    const oneInchABI = getAbi(config.liquidationMonitor.oneInchABI);
+    data.comptrollerContract = new ethers.Contract(
+      comptrollerAddress,
+      comptrollerABI,
+      data.provider,
+    );
+    data.oneInchContract = new ethers.Contract(
+      oneInchAddress,
+      oneInchABI,
+      data.provider,
+    );
+    // Unused but available for the future
+    data.oracleContract = new ethers.Contract(
+      oracleAddress,
+      oracleABI,
+      data.provider,
+    );
+
+    /* eslint-enable no-param-reassign */
+    // #endregion
+
+    // #region Get initial accounts from Compound API
+    // To-do: Can be bootstrapped with ~30 eth_getLogs calls,  500k blocks per call, using filter:
+    // Borrow (address borrower, uint256 borrowAmount, uint256 accountBorrows, uint256 totalBorrows)
+    // topic0: 0x13ed6866d4e1ee6da46f845c46d7e54120883d75c5ea9a2dacc1c4ca8984ab80
+
+    // Simple bootstrap with Compound for now
+    const apiURL = 'https://api.compound.finance/api/v2/account';
```

Comment:
This constant isn't used anywhere other than the `callAPI` function, which is imported from another file.  This should be moved to that file and removed as an argument to the `callAPI` function.  The `callAPI` function could probably also be renamed `callCompoundAPI` or something similar, as the request and response formats appear to be specific to that particular API.

Reply:
Corrected in commit bd67d88efe6427a18d45e91b638aebe719f76827